src/lib/util/Config.ts

```diff
@@ -90,6 +92,18 @@ export class Config {
 		return this.configContent.engines.find(e => e.name === name);
 	}
 
+	public async setEngineConfig(name: string, ecc: EngineConfigContent): Promise<void> {
+		// If there's an entry in the Engine array that has the same name as the one we were provided, override it with
+		// the new one.
+		const idx = this.configContent.engines.findIndex(e => e.name === name);
+		if (idx !== -1) {
+			this.configContent.engines[idx] = ecc;
+		} else {
+			this.configContent.engines.push(ecc);
```

Comment:
This line breaks the "Law of Demeter" which is a good rule to follow if possible. 
https://alvinalexander.com/java/java-law-of-demeter-java-examples/

Moving this into the ConfigContent object would solve the issue and isolate this code from changes to the ConfigContent structure.